### ISSUE TITLE
File integrity monitoring is not working as expected

### ISSUE THREAD

**Issue by cman348pwah** [osquery/osquery#6789]
# Bug report

### What operating system and version are you using?

Oracle Linux 7
version = Red Hat Enterprise Linux Server release 7.7 (Maipo)
 build =
platform = rhel

### What version of osquery are you using?

version = 4.4.0
### What steps did you take to reproduce the issue?
using File integrity monitoring pack copied directly from docs and modified to monitor an additional directory.  no file access monitoring, no excluded directories.  Here is the pack:

{
  "queries": {
    "file_events": {
      "query": "select * from file_events;",
      "removed": false,
      "interval": 300
    }
  },
  "file_paths": {
    "homes": [
      "/root/.ssh/%%",
      "/home/%/.ssh/%%"
    ],
      "etc": [
      "/etc/%%"
    ],
      "home": [
      "/home/%%"
    ],
      "tmp": [
      "/tmp/%%"
    ],
     "u01_network": [
     "/u01/app/oracle/product/19.3.0/db_1/network/admin/%%"
   ]
}
  }

osqueryctl config-check  finds no issues.  However, when i tested by purposefully modifying a file within
/u01/app/oracle/product/19.3.0/db_1/network/admin/   there was no sign of this in the osqueryd.results.log

At this point, we only want to see the FIM as we are using other tools...but here is my conf:
{
  "options": {
    "config_plugin": "filesystem",
    "logger_plugin": "filesystem",
    "logger_path": "/var/log/osquery",
    "disable_logging": "false",
    "schedule_splay_percent": "10",
    "pidfile": "/var/osquery/osquery.pidfile",
    "events_expiry": "3600",
    "database_path": "/var/osquery/osquery.db",
    "verbose": "false",
    "worker_threads": "2",
    "disable_events": "false",
    "disable_audit": "false",
    "audit_allow_config": "true",
    "host_identifier": "hostname",
    "enable_syslog": "true",
    "audit_allow_sockets": "true",
    "schedule_default_interval": "3600"
  },

#"schedule": {
#    "crontab": {
 #     "query": "SELECT * FROM crontab;",
  #    "interval": 300
  #  },
  #  "system_profile": {
  #    "query": "SELECT * FROM osquery_schedule;"
  #  },
  #  "system_info": {
  #    "query": "SELECT hostname, cpu_brand, physical_memory FROM system_info;",
  #    "interval": 3600
  #  }
  #},

 "packs": {
 #    "osquery-monitoring": "/usr/share/osquery/packs/osquery-monitoring.conf",
#"vuln-management": "/usr/share/osquery/packs/vuln-management.conf",
"fim":"/usr/share/osquery/packs/fim.conf"
 }
}

### What did you expect to see?
We expected to see evidence of the change to a file in the /u01/app/oracle/product/19.3.0/db_1/network/admin/  path in the results log
### What did you see instead?  
Nothing in the results log, searched by that path and the file that was changed.  did a restart with
osqueryctl restart
which worked fine, but when i did a status i got some weird error messages too, maybe these are why the FIM isn't working?:

 osquery]# osqueryctl status -l
Redirecting to /bin/systemctl status osqueryd.service
● osqueryd.service - The osquery Daemon
   Loaded: loaded (/usr/local/lib/systemd/system/osqueryd.service; disabled; vendor preset: disabled)
   Active: active (running) since Fri 2020-12-04 15:04:58 EST; 10min ago
  Process: 28621 ExecStartPre=/bin/sh -c if [ -f $LOCAL_PIDFILE ]; then mv $LOCAL_PIDFILE $PIDFILE; fi (code=exited, status=0/SUCCESS)
  Process: 28619 ExecStartPre=/bin/sh -c if [ ! -f $FLAG_FILE ]; then touch $FLAG_FILE; fi (code=exited, status=0/SUCCESS)
  Process: 28616 ExecStartPre=/bin/sh -c if [ ! -f $CONFIG_FILE ]; then echo {} > $CONFIG_FILE; fi (code=exited, status=0/SUCCESS)
 Main PID: 28624 (osqueryd)
   CGroup: /system.slice/osqueryd.service
           ├─28624 /usr/bin/osqueryd --flagfile /etc/osquery/osquery.flags --config_path /etc/osquery/osquery.conf
           └─28627 /usr/bin/osqueryd

osqueryd[28624]: W1204 15:05:03.059327 28627 inotify.cpp:351] Could not add inotify watch on: /home/...undles/
osqueryd[28624]: W1204 15:05:03.059394 28627 inotify.cpp:351] Could not add inotify watch on: /home/...ps/dvt/
osqueryd[28624]: W1204 15:05:03.059461 28627 inotify.cpp:351] Could not add inotify watch on: /home/.../en-US/
osqueryd[28624]: W1204 15:05:03.059530 28627 inotify.cpp:351] Could not add inotify watch on: /home/...nls/en/
osqueryd[28624]: W1204 15:05:03.059599 28627 inotify.cpp:351] Could not add inotify watch on: /home/.../chart/
osqueryd[28624]: W1204 15:05:03.060634 28627 inotify.cpp:351] Could not add inotify watch on: /root/.ssh/
osqueryd[28624]: W1204 15:05:03.060716 28627 inotify.cpp:351] Could not add inotify watch on: /home/grid/.ssh/
osqueryd[28624]: W1204 15:05:03.060732 28627 inotify.cpp:351] Could not add inotify watch on: /home/...e/.ssh/
osqueryd[28624]: W1204 15:05:03.060748 28627 inotify.cpp:351] Could not add inotify watch on: /tmp/
osqueryd[28624]: W1204 15:05:03.060766 28627 inotify.cpp:351] Could not add inotify watch on: /u01/a.../admin/

I'm very new to this...its probably something simple i hope, please advise, thank you!

**Comment by mike-myers-tob** [comment]
Yes,  I think these warnings are a hint for what's wrong. It looks like osquery was saying it could not add inotify watches on the paths you specified in your osquery config (the warnings like `osqueryd[28624]: W1204 15:05:03.060766 28627 inotify.cpp:351] Could not add inotify watch on: /u01/a.../admin/`). This table uses the iNotify subsystem in the kernel to watch files for events. It doesn't normally issue these warnings, but it can happen if osquery doesn't have enough permissions. For example if `osqueryi` is running as the local user instead of root, it can issue these warnings about root-owned files.
